TodoWithControllersAuthJWT/Services/AuthService.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections.Generic;
+using TodoWithControllersAuthJWT;
+
+namespace Todos
+{
+    internal class AuthService : IAuthService
+    {
+        private readonly Dictionary<string, (string Password, string[] Claims)> _users;
+
+        public AuthService(string issuer, byte[] key, Dictionary<string, (string Password, string[] Claims)> users)
+        {
+            Issuer = issuer;
+            Key = key;
+            _users = users;
+        }
+
+        public string Issuer { get; private set; }
+
+        public byte[] Key { get; private set; }
```

Comment:
I think the private key should be decoupled from the IAuthService. Move the key to the controller itself where it's used.

Reply:
The problem is that the key is used in app initialization (when JWT gets set up) and in controller (ie generation of tokens), imho should be in one common place.

Reply:
I just realized this, see my new comment (Sorry for the confusion)